test/lit.cfg

```diff
@@ -2400,10 +2400,15 @@ def find_compiler_rt_libs():
     # Next check for the old scheme 'clang/lib/<os-name>', ignoring
     # any target environment which is currently part of 'run_os'.
     path = make_path(base, run_os.split('-')[0])
+
+    # Check if the Android environment needs to be in the name.
+    env = ''
+    if run_os.startswith('linux-android'):
+        env = '-android'
     if os.path.exists(path):
         # We should then have the architecture in the name.
         for lib in os.listdir(path):
-            match = re.match(r'(?:lib)?clang_rt\.(\w+)-' + run_cpu, lib)
+            match = re.match(r'(?:lib)?clang_rt\.(\w+)-' + run_cpu + env, lib)
```

Comment:
If running the Android x86_64 tests on linux x86_64, this was detecting the linux clang_rt libraries and wrongly enabling all these sanitizer tests for Android too, which then fail because the compiler correctly checks for this environment suffix. Checking for the suffix here too makes sure the Android-specific libraries are available, and disables the sanitizer tests on the Android CI when they aren't.